src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -309,17 +398,17 @@ private MessageHandlerOptions DetermineMessageHandlerOptions(AzureServiceBusMess
             return messageHandlerOptions;
         }
 
-        private async Task ReAuthenticateMessageReceiverAsync(Exception exception)
+        /// <summary>
+        /// Restart core functionality of the message pump.
+        /// </summary>
+        public async Task RestartAsync()
         {
-            if (exception is UnauthorizedException)
-            {
-                Logger.LogWarning("Unable to connect anymore to Azure Service Bus, trying to re-authenticate...");
-                
-                Logger.LogTrace("Restarting Azure Service Bus...");
-                await CloseMessageReceiverAsync();
-                await OpenNewMessageReceiverAsync();
-                Logger.LogInformation("Azure Service Bus restarted!");
-            }
+            Interlocked.Exchange(ref _unauthorizedExceptionCount, 0);
+
+            Logger.LogTrace("Restarting Azure Service Bus...");
+            await CloseMessageReceiverAsync();
+            await OpenNewMessageReceiverAsync();
+            Logger.LogInformation("Azure Service Bus restarted!");
```

Comment:
I would add message pump here, and optionally more information such as the name of entity being processed, job, etc.
```suggestion
            Logger.LogInformation("Azure Service Bus message pump restarted!");
```

That would help separating pumps that are constantly restarting compared to others

Reply:
There is not entity being processed at this point, this is called upon a receival of an event or when the we receive unauthorized failures in the message pump.

Reply:
Can you elaborate on that please? I don't think I get what you're saying.

Reply:
This method where your comment is marked (`RestartAsync`) is called in two places:
1. When we receive unauthorized failures from the `MessageReceiver` itself, or
2. When we auto-restart upon the receival of `SecretNewVersionCreated` Key Vault events.

Both places don't interact with received messages. Ah, but now that I re-read your comment, you probably mean the `EntityPath` property, I thought you meant the message itself 😅 .
I'll change that!

Reply:
Resolving bc I added these extra information like the other tracing.